### ISSUE TITLE
Bun can’t run module with follow-redirects / axios bundled (`contentful-management` package)

### ISSUE THREAD

**Issue by controversial** [oven-sh/bun#2722]
tl;dr: basic usage of the `contentful-export` package throws an error from inside `axios` internals.

### What version of Bun is running?

0.6.0

### What platform is your computer?

Darwin 22.4.0 arm64 arm

### What steps can reproduce the bug?

test.ts:
```ts
import contentfulExport from 'contentful-export';

const contentModel = await contentfulExport({
  spaceId: 'abc',
  managementToken: 'abc',
});

console.log(contentModel);
```
(Note these are garbage values for `spaceId` and `managementToken`, but the same issue occurs with real values)

Run `bun test.ts`.

### What is the expected behavior?

The script runs and prints the final result (or, in the case of garbage data as above, reports a 404 for “the resource could not be found”).

### What do you see instead?

```
❯ bun test.ts
[0.17ms] ".env"
  ✔ Initialize client (1s)
  ❯ Fetching data from space
    ✖ Connecting to space (1s)
      → TypeError: socket.setTimeout is not a function. (In 'socket.setTimeout(m
secs)', 'socket.setTimeout' is undefined)…
      Fetching content types data
      Fetching tags data
      Fetching editor interfaces data
      Fetching content entries data
      Fetching assets data
      Fetching locales data
      Fetching webhooks data
      Fetching roles data
    Download assets
    Write export log file


The following 1 errors and 0 warnings occurred:

15:56:09 - Error: TypeError: socket.setTimeout is not a function. (In 'socket.setTimeout(msecs)', 'socket.setTimeout' is undefined)
```

### Additional information

`contentful-export` requires `contentful-management`, which distributes a webpack bundle called `contentful-management.node.js`.

As far as I can tell, this `socket.setTimeout` error comes from inside code bundled for the [`follow-redirects`](https://www.npmjs.com/package/follow-redirects) package, which is used internally by `axios`.

**Comment by Jarred-Sumner** [comment]
ah we are missing `socket.setTimeout` somewhere

**Comment by controversial** [comment]
I can no longer reproduce this issue on latest canary
